src/components/TextInputs/__tests__/NumberInput.test.tsx

```diff
@@ -0,0 +1,95 @@
+import React from 'react'
+import { TextInput as RNTextInput } from 'react-native'
+import renderer, { ReactTestRenderer } from 'react-test-renderer'
+import NumberInput from '../NumberInput'
+
+const mockPlatform = (OS: string, version: number) => {
+  jest.resetModules()
+  const selectedOS = [OS]
+  jest.doMock('Platform', () => ({
+    OS,
+    select: selectedOS,
+    Version: version || undefined,
+  }))
+}
+
+describe('NumberInput', () => {
+  let testRenderer: ReactTestRenderer
+
+  beforeEach(() => {
+    const onChangeText = jest.fn()
+    testRenderer = renderer.create(<NumberInput onChangeText={onChangeText} />)
+  })
+
+  describe('when changing focus', () => {
+    it('renders with gray border when blurred', () => {
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onBlur()
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+
+    it('renders with green border when focused', () => {
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onFocus()
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+  })
+
+  describe('when adding or updating text', () => {
+    it('renders a valid digit', () => {
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onChangeText('5')
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+
+    it('does not render an invalid digit', () => {
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onChangeText('B')
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+  })
+
+  describe('when key is pressed', () => {
+    it('renders a valid digit', () => {
+      const input = testRenderer.root.findByType(RNTextInput)
+      input.props.onKeyPress({ nativeEvent: { key: '5' } })
+      expect(testRenderer.toJSON()).toMatchSnapshot()
+    })
+
+    it('does not render an invalid digit', () => {
```

Comment:
I'm not sure about these tests. In this case, for example, it could be rendering `B` and we're not asserting anything related to this. What will happen is that, next time this test run, it will pass because the same error happened. What do you think, @barbosa?

Reply:
Yeah. Maybe we should check for the values of the inputs themselves instead of relying on snapshots in this case.

Reply:
Agree!